### PR TITLE
Add json formatted output for ENV printing.

### PR DESCRIPTION
**Pull request by tpickett66**

### CODE REVIEW

File: args_test.go [hunk 1]
```diff
@@ -383,9 +394,6 @@ var (
 		{
 			Args: []string{"env"},
 		},
-		{
```
Comment (miquella) [inline args_test.go]: This should be put back, this still isn't allowed.

File: env.go [hunk 2]
```diff
@@ -1,16 +1,59 @@
 package main
 
 import (
```
Comment (miquella) [inline env.go]: Need to run `go fmt` :)

File: env.go [hunk 3]
```diff
@@ -19,47 +62,32 @@ func (e *Env) Run(steward Steward) error {
...
-
-	// sort the vars to unset
-	sort.Strings(vars.Unset)
+	templateStr := envFormatters["sh"]
 
-	// sort the vars to set
-	var keys []string
-	for key := range vars.Set {
-		keys = append(keys, key)
+	if e.Format == "shell" {
```
Comment (miquella) [inline env.go]: I think something like this is a little more straight forward:
```
	format := e.Format
	if format == "shell" {
		format = e.DetectedShell
	}

	if thing, ok := envFormatters[format]; ok {
		templateStr = thing
	} else {
		templateStr = format
	}
```

File: env.go [hunk 4]
```diff
@@ -19,47 +62,32 @@ func (e *Env) Run(steward Steward) error {
...
-		fmt.Printf(setVar, key, strings.Replace(vars.Set[key], "\"", quoteReplacement, -1))
-	}
+	unsets := variables.Unset
+	sort.Strings(unsets)
+	vals.Unset = unsets
+
+	vals.Command = e.Command
 
-	return nil
+	tmpl, err := template.New("envTmpl").Funcs(templateFuncMap).Parse(templateStr)
```
Comment (miquella) [inline env.go]: I _might_ suggest hoisting this to where the format template is determined (before the creation of the `templateVals`).

File: env_test.go [hunk 5]
```diff
@@ -1,24 +1,23 @@
 package main
 
 import (
-	"os"
 	"testing"
+	"fmt"
 
 	"github.com/miquella/vaulted/lib"
+	"os"
```
Comment (miquella) [inline env_test.go]: Please keep golang packages together (and separate from external libs).

File: env_test.go [hunk 6]
```diff
@@ -1,24 +1,23 @@
...
-	envFishOutputWithHint = `# To load these variables into your shell, execute:
+	envFishOutput = `# To load these variables into your shell, execute:
 #   eval (vaulted env one)
 set -x ONE "111111";
 set -x THREE "333";
 set -x TWO "222";
 `
-	envFishOutputWithPermCreds = `set -e AWS_SECURITY_TOKEN;
+	envFishOutputWithPermCreds = `# To load these variables into your shell, execute:
+#   eval ()
```
Comment (miquella) [inline env_test.go]: Please reintroduce the `Command` into this test :)

File: env_test.go [hunk 7]
```diff
@@ -27,27 +26,33 @@ set -x THREE "333";
...
-	envShOutputWithHint = `# To load these variables into your shell, execute:
+	envShOutput = `# To load these variables into your shell, execute:
 #   eval $(vaulted env one)
 export ONE="111111"
 export THREE="333"
 export TWO="222"
 `
-	envShOutputWithPermCreds = `unset AWS_SECURITY_TOKEN
+	envShOutputWithPermCreds = `# To load these variables into your shell, execute:
+#   eval $()
```
Comment (miquella) [inline env_test.go]: And here

File: args.go [hunk 8]
```diff
@@ -255,18 +256,13 @@ func parseEnvArgs(args []string) (Command, error) {
...
-	}
-
 	e := &Env{}
 	e.VaultName = flag.Arg(0)
-	e.Shell = shell
-	e.UsageHint = usageHint
+	e.DetectedShell = shell
+	e.Command = fmt.Sprintf("vaulted env %s", strings.Join(args, " "))
+
+	format, _ := flag.GetString("format")
```
Comment (tpickett66) [inline args.go]: ✅

File: args_test.go (hunk 1 above)
Comment (tpickett66) [inline args_test.go]: ✅

File: env.go (hunk 2 above)
Comment (tpickett66) [inline env.go]: ✅

File: env.go (hunk 3 above)
Comment (tpickett66) [inline env.go]: ✅

File: env.go (hunk 4 above)
Comment (tpickett66) [inline env.go]: ✅

File: env_test.go (hunk 5 above)
Comment (tpickett66) [inline env_test.go]: ✅

File: env_test.go (hunk 6 above)
Comment (tpickett66) [inline env_test.go]: ✅

File: env_test.go (hunk 7 above)
Comment (tpickett66) [inline env_test.go]: ✅

File: env.go [hunk 9]
```diff
@@ -19,47 +59,32 @@ func (e *Env) Run(steward Steward) error {
...
-	if e.UsageHint {
-		fmt.Printf(usageHint, strings.Join(os.Args, " "))
-	}
+	sort.Strings(variables.Unset)
+	vals.Unset = variables.Unset
 
-	for _, key := range vars.Unset {
-		fmt.Printf(unsetVar, key)
-	}
+	vals.Command = strings.Join(os.Args, " ")
```
Comment (miquella) [inline env.go]: I believe you had a previous patchset that passed the `Command` into the `Env` object, that seems more correct to me. So, I think we should add `Command` back to the `Env` struct and move the `strings.Join(os.Args, " ")` out to the arg parser (where I should have put it to begin with…).

File: args.go [hunk 10]
```diff
@@ -255,18 +256,12 @@ func parseEnvArgs(args []string) (Command, error) {
...
-			usageHint = false
-		}
-	}
-
 	e := &Env{}
 	e.VaultName = flag.Arg(0)
-	e.Shell = shell
-	e.UsageHint = usageHint
+	e.DetectedShell = shell
+	e.DetectedShell = shell
```
Comment (miquella) [inline args.go]: I'm mostly sure we don't need to assign `.DetectedShell` twice 😉

File: env.go (hunk 9 above)
Comment (tpickett66) [inline env.go]: ... 😒

**Pull request merged**